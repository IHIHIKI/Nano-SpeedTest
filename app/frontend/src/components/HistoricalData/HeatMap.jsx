import React from 'react';
import PropTypes from 'prop-types';
import 'styles/GoogleMaps.css';

const base = 'https://maps.googleapis.com/maps/api/staticmap';
const scale = '2';
const size = `1200x600`;
const googleKey = process.env.REACT_APP_GOOGLE_MAPS_KEY;

const HeatMap = ({nodeLocations}) => {
    const points = nodeLocations.map((node) => node.coords );
    const markers = `size:small|${points.join('|')}`;
    // Google maps will auto zoom and fit any markers on a map
    return (
        <img className='GoogleHeatMap'
        src={`${base}?size=${size}&scale=${scale}&markers=${markers}&key=${googleKey}`}
        alt={'Google Map Showing Locations of Nodes'} />
    );
};

HeatMap.propTypes = {
    nodeLocations: PropTypes.array.isRequired
};

export default HeatMap;
