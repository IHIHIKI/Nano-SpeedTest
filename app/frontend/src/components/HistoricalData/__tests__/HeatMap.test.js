import React from 'react';
import renderer from 'react-test-renderer';
import HeatMap from '../HeatMap';
import { shallow } from 'enzyme';

const base = 'https://maps.googleapis.com/maps/api/staticmap';
const scale = '2';
const size = `1200x600`;
const googleKey = process.env.REACT_APP_GOOGLE_MAPS_KEY;
let points;
let markers;

it('shallow renders without crashing', () => {
    const nodeLocations = {
        444: {
            latitude: 123,
            longitude: 456,
            coords: '123,456'
        }
    };
    shallow(<HeatMap nodeLocations={nodeLocations}/>);
    points = Object.keys(nodeLocations).map((id) => nodeLocations[id].coords);
    markers = `size:small|${points.join('|')}`;
});

it(`renders snapshot correctly`, () => {

    const tree = renderer.create(
        <img className='GoogleHeatMap'
             src={`${base}?size=${size}&scale=${scale}&markers=${markers}&key=${googleKey}`}
             alt={'Google Map Showing Locations of Nodes'} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});