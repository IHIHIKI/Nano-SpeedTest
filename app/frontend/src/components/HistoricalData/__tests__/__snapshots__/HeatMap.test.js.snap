// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<img
  alt="Google Map Showing Locations of Nodes"
  className="GoogleHeatMap"
  src="https://maps.googleapis.com/maps/api/staticmap?size=1200x600&scale=2&markers=size:small|123,456&key=AIzaSyAovwwttuLBsvolc97QUc5pejsbwfLPPZs"
/>
`;
